packages/components/src/Modal/GenericModal/GenericModal.tsx

```diff
@@ -1,71 +1,45 @@
-import React, { useEffect, useId, useState } from 'react'
+import React, { useId, useRef } from 'react'
 import { createPortal } from 'react-dom'
-import { Transition } from '@headlessui/react'
-import FocusLock from 'react-focus-lock'
+import classNames from 'classnames'
+import { AnimatePresence, motion } from 'motion/react'
+import { FocusOn } from 'react-focus-on'
 import { warn } from '../util/console'
+import { backdropMotion, modalMotion } from './animations'
 import { ModalContext } from './context/ModalContext'
-import styles from './GenericModal.module.scss'
+import styles from './GenericModal.module.css'
 
 export type GenericModalProps = {
   id?: string
   isOpen: boolean
   children: React.ReactNode
-  focusLockDisabled?: boolean
-  onEscapeKeyup?: (event: KeyboardEvent) => void
+  onEscapeKeyup?: (event: Event) => void
   onOutsideModalClick?: (event: React.MouseEvent) => void
   /** A callback that is triggered after the modal is opened. */
   onAfterEnter?: () => void
   /** A callback that is triggered after the modal is closed. */
   onAfterLeave?: () => void
   className?: string
+  size?: 'medium' | 'large'
```

Comment:
Could you add a type description for this just to outline it controls the visual size of the container